src/serac/physics/integrators/hyperelastic_traction_integrator.hpp

```diff
@@ -0,0 +1,173 @@
+
+/**
+ * @brief Custom MFEM integrator for nonlinear finite deformation traction loads
+ */
+class HyperelasticTractionIntegrator : public mfem::NonlinearFormIntegrator {
+public:
+  /**
+   * @brief Construct a new Hyperelastic Traction Integrator object
+   *
+   * @param[in] f The traction load coefficient
+   * @param[in] compute_on_reference Flag for computing the residual in the stress-free configuration vs. the deformed
+   * configuration
+   */
+  explicit HyperelasticTractionIntegrator(mfem::VectorCoefficient& f, bool compute_on_reference = false)
+      : function_(f), compute_on_reference_(compute_on_reference)
+  {
+  }
+
+  /**
+   * @brief Assemble the nonlinear residual on a boundary element at a current state
+   *
+   * @param[in] el1 The first element attached to the face
+   * @param[in] el2 The second element attached to the face
+   * @param[in] Tr The face element transformation
+   * @param[in] elfun The current value of the underlying finite element state for residual evaluation
+   * @param[out] elvec The evaluated residual
+   */
+  virtual void AssembleFaceVector(const mfem::FiniteElement& el1, const mfem::FiniteElement& el2,
+                                  mfem::FaceElementTransformations& Tr, const mfem::Vector& elfun, mfem::Vector& elvec);
+
+  /**
+   * @brief Assemble the gradient for the nonlinear residual at a current state
+   *
+   * @param[in] el1 The first element attached to the face
+   * @param[in] el2 The second element attached to the face
+   * @param[in] Tr The face element transformation
+   * @param[in] elfun The current value of the underlying finite element state for gradient evaluation
+   * @param[out] elmat The local contribution to the Jacobian
+   */
+  virtual void AssembleFaceGrad(const mfem::FiniteElement& el1, const mfem::FiniteElement& el2,
+                                mfem::FaceElementTransformations& Tr, const mfem::Vector& elfun,
+                                mfem::DenseMatrix& elmat);
+
+  /**
+   * @brief Destroy the Hyperelastic Traction Integrator object
+   */
+  virtual ~HyperelasticTractionIntegrator() {}
+
+private:
+  /**
+   * @brief The vector coefficient for the traction load
+   */
+  mfem::VectorCoefficient& function_;
+
+  /**
+   * @brief Working matrices for the traction calculations
+   */
+  mutable mfem::DenseMatrix DSh_u_;
+  mutable mfem::DenseMatrix DS_u_;
+  mutable mfem::DenseMatrix J0i_;
+  mutable mfem::DenseMatrix F_;
+  mutable mfem::DenseMatrix Finv_;
+  mutable mfem::DenseMatrix FinvT_;
+  mutable mfem::DenseMatrix PMatI_u_;
+
+  /**
+   * @brief Working vectors for the traction
+   */
+  mutable mfem::Vector shape_;
+  mutable mfem::Vector nor_;
+  mutable mfem::Vector fnor_;
+  mutable mfem::Vector Sh_p_;
+  mutable mfem::Vector Sh_u_;
+
+  bool compute_on_reference_;
+};
+
+/**
+ * @brief Custom MFEM integrator for pressure loads
+ */
+class HyperelasticPressureIntegrator : public mfem::NonlinearFormIntegrator {
```

Comment:
what does "Hyperelastic" mean in the context of tractions/pressures?

Reply:
Sam makes a good point.  I think the only two possibilities for the RHS are "linear (normals/areas fixed) and nonlinear".  In the latter case one can specify variations on the tangent, whether you want to include it at all (noting that the consistent version is nonsymmetric)

Reply:
It means nothing other than I developed them at the same time.  😄 I'll remove it.